aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1511,7 +1511,12 @@ impl AptosVM {
         new_published_modules_loaded: &mut bool,
         change_set_configs: &ChangeSetConfigs,
     ) -> Result<UserSessionChangeSet, VMStatus> {
-        let maybe_publish_request = session.execute(|session| session.extract_publish_request());
+        let maybe_publish_request = session.execute(|session| {
+            // With V2 loader, do not allow init_module to request module publish. While allowing
+            // it is harmless, the user experience can be bad - so we better avoid it completely.
+            let disable_publishing = self.features().is_loader_v2_enabled();
```

Comment:
Are we assuming that this is not a breaking change because not used (replay)? Or do v1 and v2 actually behave different, so before the feature is enabled we use v1. The later is a bit problematic because we want to remove the v1 code, so v2 should be able to replay full history, but perhaps has a v1_compatible flag.

Reply:
My take was that we can bundle it together with the loader as it is not used, but maybe it is safer to usa a feature flag for this straight away. Added a new feature flag.